### PR TITLE
Don't generate #[doc(hidden)] elements

### LINKED ISSUE #34025: Rustdoc should not generate docs for hidden methods.
See previous discussion here: https://github.com/rust-lang/rust/issues/34023#issuecomment-223094149

rustdoc seems to be building files for methods marked `#[doc(hidden)]` in rust 1.9.0 and nightly but not in 1.8.0, observed with code generated when building [opencv-rust](https://github.com/kali/opencv-rust).
It seems that these methods should be ignored.

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
Fixes #34025.

r? @brson

### CODE REVIEW

File: src/test/rustdoc/issue-15347.rs [hunk 1]
```diff
@@ -1,15 +0,0 @@
...
-// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
-// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
-// option. This file may not be copied, modified, or distributed
-// except according to those terms.
-
-// compile-flags:--no-defaults --passes collapse-docs --passes unindent-comments
-
-// @has issue_15347/fn.foo.html
-#[doc(hidden)]
-pub fn foo() {}
```
Comment (brson) [inline src/test/rustdoc/issue-15347.rs]: I don't think this test should be deleted. From #15347 it seems that these compile flags are supposed to result in this file existing, *with actual content* (whereas this PR is removing empty files for hidden docs).

File: src/librustdoc/html/render.rs [hunk 2]
```diff
@@ -1369,7 +1369,9 @@ impl Context {
             self.render_redirect_pages = self.maybe_ignore_item(&item);
         }
 
-        if item.is_mod() {
+        if item.attrs.list("doc").has_word("hidden") {
+            Ok(())
+        } else if item.is_mod() {
```
Comment (brson) [inline src/librustdoc/html/render.rs]: It seems that the strip_hidden pass is supposed to be responsible for removing hidden items. Why is that pass not sufficient? Shouldn't the hidden items have been stripped by this point and not be seen here?

File: src/test/rustdoc/issue-15347.rs (hunk 1 above)
Comment (GuillaumeGomez) [inline src/test/rustdoc/issue-15347.rs]: The thing here is that the `foo` function is the only item and is specified as hidden. So we have nothing to generate. If I understood correctly, #15347 was about creating doc for *private* items, not *hidden* ones.

File: src/librustdoc/html/render.rs (hunk 2 above)
Comment (GuillaumeGomez) [inline src/librustdoc/html/render.rs]: Like I said, the pass only set them as hidden but still generates the files. This change prevent the creation of the files.

**Comment by GuillaumeGomez** [comment]
Closing it in favor of #34045.

**Pull request closed without merging**